scripts/kconfig/kconfig.py

```diff
@@ -71,6 +48,39 @@ def name_and_loc(sym):
     kconf.write_autoconf(args.autoconf)
 
 
+def verify_assigned_value(sym):
+    # Verifies that the value assigned to 'sym' "took" (matches the value the
+    # symbol actually got), printing a warning otherwise
+
+    # Tristate values are represented as 0, 1, 2. Having them as
+    # "n", "m", "y" is more convenient here, so convert.
+    if sym.type in (BOOL, TRISTATE):
+        user_value = TRI_TO_STR[sym.user_value]
+    else:
+        user_value = sym.user_value
+
+    if user_value != sym.str_value:
+        print('warning: {} was assigned the value "{}" but got the value "{}". '
+              "Check its dependencies at "
+              "http://docs.zephyrproject.org/reference/kconfig/CONFIG_{}.html "
+              "or in the 'menuconfig' interface (see the Application "
+              "Development Primer section of the manual)."
```

Comment:
Checking the dependencies on the website seems like a bad idea.

When the Kconfig is different on the website from locally you will have a bad time.

Perhaps just suggest checking menuconfig.

Reply:
Yeah, it might be outdated if someone changed something very recently. It will probably be correct in 90% of cases though, and I think it would help a lot more people find it.

What about adding a hint that it might be outdated?

Reply:
Maybe something like "...check its dependencies in the 'menuconfig' interface (see the ...) or in the symbol reference at http://..., which is updated daily"?

Reply:
It doesn't need to be recently. People can be working on forks that are several years old
or permanent forks with slightly different Kconfig.

I wouldn't use this, as I don't like tools that are correct 90% of the time.

But if you believe this is useful then this is OK by me.